api/v1/db/json/notes-dao.js

```diff
@@ -0,0 +1,93 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+const fs = require('fs');
+const config = require('config');
+
+const { serializeNotes } = require('../../serializers/notes-serializer');
+
+/**
+ * @summary Return a list of notes filtered/sorted by query parameters
+ * @function
+ * @param {Object} query Query parameters
+ * @returns {Promise} Promise object represents a list of notes
+ */
+const getNotes = query => new Promise((resolve, reject) => {
+  try {
+    const { dbDirectoryPath } = config.api;
+    if (!fs.existsSync(`${appRoot}${dbDirectoryPath}`)) {
+      reject(new Error(`DB directory path: '${dbDirectoryPath}' is invalid`));
+    }
+
+    const {
+      studentID, q, sortKey, sources, contextTypes,
+    } = query;
+    const dirPath = `${dbDirectoryPath}/${studentID}`;
+    let files;
+    let rawNotes = [];
+    try {
+      files = fs.readdirSync(`${appRoot}${dirPath}`);
+    } catch (ignore) {
+      // rawNotes should remain an empty array if directory does not exist
+    }
+
+    _.forEach(files, (file) => {
+      rawNotes.push(appRoot.require(`${dirPath}/${file}`));
+    });
+
+    if (contextTypes) {
+      const contexts = contextTypes.toString().split(',');
+      rawNotes = _.filter(rawNotes, it => _.includes(contexts, it.context.contextType));
+    }
+
+    if (q) {
+      rawNotes = _.filter(rawNotes, it => _.includes(it.note, q));
+    }
+
+    // sort first by sortKey, and then by lastModified within each sorted group
+    rawNotes = _.orderBy(
+      rawNotes,
+      [
+        sortKey.toString() === 'contextType' ? it => it.context[sortKey] : it => it[sortKey],
+        it => it.lastModified,
+      ],
+      [sortKey.toString() === 'lastModified' ? 'desc' : 'asc', 'desc'],
+    );
+
+    _.forEach(rawNotes, (it) => {
+      it.source = 'advisorPortal';
+    });
+
+    if (sources) {
+      const sourcesList = sources.toString().split(',');
```

Comment:
This logic appears at least twice. Thinking about it more, I think we should also move this logic into middleware somewhere to parse any comma-separated path parameters into an array.

Reply:
I did some testing and it seems that these comma-separated path parameters are already being parsed as arrays. There's an [`enableObjectCoercion`](https://github.com/kogosoftwarellc/open-api/tree/master/packages/express-openapi#argsenableobjectcoercion) argument in the `initialize` function that I assumed would need to be enabled to correctly parse these parameters but I couldn't see any difference with this enabled or disabled.